vllm/config.py

```diff
@@ -1794,12 +1794,119 @@ def __init__(self, device: str = "auto") -> None:
             self.device = torch.device(self.device_type)
 
 
+@dataclass
 class SpeculativeConfig:
-    """Configuration for speculative decoding.
+    """
+    Configuration for speculative decoding.
+    Configurable parameters include:
+    - Top-level Speculative Decoding Control:
+        - num_speculative_tokens (int): The number of speculative
+            tokens, if provided. It will default to the number in the draft
+            model config if present, otherwise, it is required.
+        - proposer (Optional[str]): The name of the speculative method to use.
+            Defaults to the model name if not provided.
```

Comment:
This is a bit confusing. I guess you're saying if not provided, we assume the proposer is the draft model?